addons/account/models/account_bank_statement.py

```diff
@@ -199,7 +199,7 @@ def _get_previous_statement(self):
             # not have any id yet. However if we are updating an existing record, the domain date <= st.date
             # will find the record itself, so we have to add a condition in the search to ignore self.id
             if not isinstance(st.id, models.NewId):
-                domain.append(('id', '!=', st.id))
+                domain.extend(['|', '&', ('id', '<', st.id), ('date', '=', st.date), '&', ('id', '!=', st.id), ('date', '!=', st.date)])
```

Comment:
Shouldn't this line that belong to its own commit be merged in saas-13.3 instead of this branch?

Reply:
Yes, I will backport it if you are ok with it but I needed it here (because of the test, but I could change that)

Reply:
Same for the second commit of this branch, both should be backported